iree/compiler/Dialect/Flow/IR/PartitionableLoopsInterface.cpp

```diff
@@ -0,0 +1,197 @@
+  unsigned getNumLoops(Operation *op) const {
+    auto linalgOp = cast<linalg::LinalgOp>(op);
+    return linalgOp.getNumLoops();
+  }
+
+  SmallVector<unsigned> getPartitionableLoops(
+      Operation *op, unsigned maxNumPartitionedLoops) const {
+    return {0, 1};
+  }
+};
+
+/// External model implementation for all operations that implement the
+/// `TiledOpInterface`.
+struct TiledOpInterfacePartitionableLoops
+    : public PartitionableLoopsInterface::ExternalModel<
+          TiledOpInterfacePartitionableLoops, LinalgExt::TiledOpInterface> {
+  unsigned getNumLoops(Operation *op) const {
+    auto tiledOp = cast<LinalgExt::TiledOpInterface>(op);
+    return tiledOp.getLoopIteratorTypes().size();
+  }
+
+  SmallVector<unsigned> getPartitionableLoops(
+      Operation *op, unsigned maxNumPartitionedLoops) const {
+    // For now just return the loops that are returned by the
+    // `TiledOpInterface`. This needs to be further pruned to remove unit-dim
+    // loops, but that needs the interface to return the static sizes of the
+    // loops.
+    auto tiledOp = cast<LinalgExt::TiledOpInterface>(op);
+    return tiledOp.getPartitionableLoops(maxNumPartitionedLoops);
+  }
+};
+
+/// Registers the `LinalgOpPartitionableLoops` model for all Linalg ops. This
+/// needs to be done on a op-by-op basis since registration is on an op-by-op
+/// basis.
+template <typename OpTy>
+static void registerInterfaceForLinalgOps(DialectRegistry &registry) {
+  registry.addOpInterface<OpTy, LinalgOpPartitionableLoops>();
+}
+
+/// Specializations of the registration method to use a different external model
+/// instead of the generic external model for Linalg ops.
+template <>
+void registerInterfaceForLinalgOps<linalg::Mmt4DOp>(DialectRegistry &registry) {
+  registry.addOpInterface<linalg::Mmt4DOp, Mmt4DOpPartitionableLoops>();
+}
+
+/// Registers the external models for all Linalg operations.
+template <typename OpTy1, typename OpTy2, typename... More>
+static void registerInterfaceForLinalgOps(DialectRegistry &registry) {
+  registerInterfaceForLinalgOps<OpTy1>(registry);
+  registerInterfaceForLinalgOps<OpTy2, More...>(registry);
+}
+
+/// Registers the `TiledOpInterfacePartitionableLoops` model for operations.
+template <typename OpTy>
+static void registerInterfaceForTiledOpInterfaceOps(DialectRegistry &registry) {
+  registry.addOpInterface<OpTy, TiledOpInterfacePartitionableLoops>();
+}
+
+/// Registers the external models for all TiledOpInterface operations.
+template <typename OpTy1, typename OpTy2, typename... More>
+static void registerInterfaceForTiledOpInterfaceOps(DialectRegistry &registry) {
+  registerInterfaceForTiledOpInterfaceOps<OpTy1>(registry);
+  registerInterfaceForTiledOpInterfaceOps<OpTy2, More...>(registry);
+}
+
+void registerPartitionableLoopsInterfaceModels(DialectRegistry &registry) {
+  // clang-format off
+  registerInterfaceForLinalgOps<
+  
+  // This is copy-pasted from LinalgStructuredOps.cpp.inc. In theory you could
```

Comment:
Oh yeah this can go stale and cause confusion. Curious, what's the linking error? Did you properly `#define` stuff before `#include`?